### ISSUE TITLE
xrdcp from 4.7.0 cannot write to dCache

### ISSUE THREAD

**Issue by costing** [xrootd/xrootd#593]
Hi,

Upgrading the client from 4.6.1 to 4.7.0 broke ALICE's raw data replication to T1s running dCache. The client (both with and without TPC) time out and exit with code 52 and print an "[ERROR] Operation expired". The targets run various dCache versions (2.16.30, 2.16.42, 3.0.28). Do you know what could cause this in the latest version? Transfers to Xrootd targets was not affected and we have reverted to 4.6.1 until this is understood.

Cheers,

.costin

**Comment by IATkachenko** [comment]
From site point of view for dCache 2.16.30 we have following log records

Door:
```
2017-09-29T04:05:13.708 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] Opening //t1.grid.kiae.ru/data/alice/alicetape/15/32109/c174ac1a-a0dd-11e7-9717-0c39a10abeef for write
2017-09-29T04:05:13.708 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] open flags: options to apply for open path (raw=5474 ): kXR_async kXR_delete kXR_open_updt kXR_mkpath kXR_retstat
2017-09-29T04:05:13.708 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] mode to apply to open path: rw- r-- r--
2017-09-29T04:05:13.708 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] Status: PnfsManager: Creating name space entry
2017-09-29T04:05:13.720 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] Status: PoolManager: Selecting pool
2017-09-29T04:05:13.725 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] Status: Pool tcd6-0-alicetape: Creating mover
2017-09-29T04:05:13.791 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] Status: Mover tcd6-0-alicetape/98427: Waiting for redirect
2017-09-29T04:05:16.113 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] Status: Mover tcd6-0-alicetape/98427: Receiving
2017-09-29T04:05:16.113 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] Redirecting to /144.206.236.195:34928
2017-09-29T04:05:16.113 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] Sending the following host information to the client: 144.206.236.195
2017-09-29T04:05:16.113 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] [id: 0x2d2dc0de, L:/10.1.0.99:1094 - R:/188.184.2.11:45850] WRITE: redirect[144.206.236.195:34928,&org.dcache.uuid=b870e58e-c5f8-471d-8bc3-0061ef83980e,]
2017-09-29T04:05:16.113 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg] [id: 0x2d2dc0de, L:/10.1.0.99:1094 - R:/188.184.2.11:45850] FLUSH
2017-09-29T04:10:16.191 (Xrootd-tape-sdrm) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg tcd6-0-alicetape DoorTransferFinished 000047DB8C06FCBC40FB9533874F225BB467] Transfer 000047DB8C06FCBC40FB9533874F225BB467@tcd6-0-alicetape failed: No connection from client after 300 seconds. Giving up. (error code=10006)
```

Pool
```
2017-09-29T04:05:13.724 (tcd6-0-alicetape) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg Xrootd-tape-sdrm PoolAcceptFile 000047DB8C06FCBC40FB9533874F225BB467] Creating new entry for 000047DB8C06FCBC40FB9533874F225BB467
... nothing about this transfer and pnfsid...
2017-09-29T04:10:16.108 (tcd6-0-alicetape) [door:Xrootd-tape-sdrm@xrootdDomain:AAVaSWS2fXg Xrootd-tape-sdrm PoolAcceptFile 000047DB8C06FCBC40FB9533874F225BB467] Transfer failed: No connection from client after 300 seconds. Giving up.

```

**Comment by simonmichal** [comment]
We are aware of this problem. Starting with 4.7.0 the xrootd client started enforcing correct kXR_login response format. Unfortunately, it turned out that in case of dCache pool nodes the response is in incorrect (the sessionId is missing).

We are planning to release 4.7.1 which will include a 'hack' for dCache sites.

There is also some effort on dCache side to fix the problem (https://github.com/dCache/dcache/pull/3562) 

Cheers,
Michal

**Comment by simonmichal** [comment]
This issue has been fixed in ed86abbaf518b0b9256f77a853ede198b385bb2c (tested at BNL).
